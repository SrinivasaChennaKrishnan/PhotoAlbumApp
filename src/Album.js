import React, { lazy, Suspense } from "react";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import Classes from "./Album.module.css";

/* Dynamic import of Component */
const PhotoAlbum = lazy(() => import("./Components/PhotoAlbum"));

/** Root Component */
function Album() {
  return (
    <div className={Classes.Album}>
      <Router>
        <Suspense fallback={<div>Loading...</div>}>
          <Switch>
            <Route exact path="/photoAlbum" component={PhotoAlbum} />
            <Redirect exact from="/" to="photoAlbum" />
          </Switch>
        </Suspense>
      </Router>
    </div>
  );
}

export default Album;
