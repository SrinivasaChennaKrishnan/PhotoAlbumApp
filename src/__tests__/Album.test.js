import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Album from '../Album'

describe('Album', () => {
  
  /** testing whether the Album component renders without crashing */
  it('renders without crashing given the required props', () => {
    const props = {
      albumData: [],
      photolistObject: [],
      loader: false,
      albumMessage: "INITIAL",
      photoListMessage: "INITIAL",
      picSource: "",
      showPhoto: false,
      selectedAlbum: [],
      selectedPhoto: [],
      dispatch: jest.fn()
    }
    const wrapper = shallow(<Album {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})