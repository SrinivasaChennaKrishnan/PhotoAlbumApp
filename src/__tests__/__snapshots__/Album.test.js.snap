// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders without crashing given the required props 1`] = `
<div
  className="Album"
>
  <BrowserRouter>
    <Suspense
      fallback={
        <div>
          Loading...
        </div>
      }
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/photoAlbum"
        />
        <Redirect
          exact={true}
          from="/"
          to="photoAlbum"
        />
      </Switch>
    </Suspense>
  </BrowserRouter>
</div>
`;
